### DISCUSSION TITLE
Is pretrained model downloaded or just remotely call via API?

### DISCUSSION THREAD

**Discussion by peiyaoli** [datamol-io/molfeat#44] [Q&A] [answered]
Hi, sorry I am new to pretrained LLM.

for the demo code:

```python
import datamol as dm
from molfeat.trans.pretrained.hf_transformers import PretrainedHFTransformer

data = dm.freesolv()
transformer = PretrainedHFTransformer(kind="ChemGPT-1.2B", notation="selfies", device="cuda:0")
features = transformer(data["smiles"])
```
this transformer model is downloaded locally or call the HF API to get the embedded feature back?

Many thanks

**Comment by maclandrol** [comment, marked as answer]
Hi @peiyaoli, all pretrained models are downloaded and cached locally for efficiency. The first time you call the pretrained model, it would take some time because of that. 

In the future, there could be models that are used through an API, A bit similar to what has been done in this plugin: [molfeat-hype](https://github.com/maclandrol/molfeat-hype) where OpenAI text embeddings are used to investigate LLM-based molecular representation.

**Reply by peiyaoli** [reply]
Hi, @maclandrol , thanks for the reply.

I am curious that why this model size of this 1.2B version is only 147M. On the HF, I saw the model size is over 4GB. Is there any compression technique you used?

**Reply by maclandrol** [reply]
You are right @peiyaoli . This is suspicious and It turns out that I have overwritten the upload with the 19M weights  😬. 

It's fixed.  In the future, I will upload all `ETL scripts` so this process is more transparent and less prone to errors. 

Please refer to this comment: https://github.com/datamol-io/molfeat/issues/29#issuecomment-1502222450  for updating your local cache "properly".

Many thanks !
